Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/consent/SubmissionConsentViewModel.kt

```diff
@@ -42,12 +55,50 @@ class SubmissionConsentViewModel @AssistedInject constructor(
                     routeToScreen.postValue(SubmissionNavigationEvents.ResolvePlayServicesException(exception))
                 } else {
                     Timber.d(exception, "Pre-auth failed with unrecoverable exception")
-                    routeToScreen.postValue(SubmissionNavigationEvents.NavigateToQRCodeScan)
+                    proceed()
                 }
             }
         }
     }
 
+    private fun proceed() {
+        qrCode.let {
+            if (it == null)
+                routeToScreen.postValue(SubmissionNavigationEvents.NavigateToQRCodeScan)
+            else
+                processQrCode(it)
+        }
+    }
+
+    private fun processQrCode(qrCodeString: String) {
+        launch {
+            validateAndRegister(qrCodeString)
+        }
+    }
+
+    private suspend fun validateAndRegister(qrCodeString: String) {
+        try {
+            val coronaTestQRCode = qrCodeValidator.validate(qrCodeString)
+            // TODO this needs to be adapted to work for different types
+            QRCodeCensor.lastGUID = coronaTestQRCode.registrationIdentifier
+            qrCodeValidationState.postValue(QrCodeRegistrationStateProcessor.ValidationState.SUCCESS)
+            val coronaTest = submissionRepository.testForType(coronaTestQRCode.type).first()
+
+            if (coronaTest != null) {
+                routeToScreen.postValue(
+                    SubmissionNavigationEvents.NavigateToDeletionWarningFragmentFromQrCode(
+                        coronaTestQRCode,
+                        consentGiven = true
+                    )
+                )
+            } else {
+                qrCodeRegistrationStateProcessor.startQrCodeRegistration(coronaTestQRCode, true)
+            }
+        } catch (err: InvalidQRCodeException) {
+            qrCodeValidationState.postValue(QrCodeRegistrationStateProcessor.ValidationState.INVALID)
+        }
+    }
```

Comment:
Please add some logs to help us identify the behaviour if we got some issues?

Reply:
There is a log statement and the validator also logs errors